hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/AbstractCSQueue.java

```diff
@@ -463,12 +463,14 @@ protected void setDynamicQueueProperties() {
           queueContext.getQueueManager().getConfiguredNodeLabelsForAllQueues()
               .setLabelsByQueue(getQueuePath(), new HashSet<>(parentNodeLabels));
         }
+        Map<String, QueueCapacityVector> templateConfiguredResourceVector = queueContext.getConfiguration()
+                .parseConfiguredResourceVector(queuePath.getFullPath(), parentNodeLabels);
+
```

Comment:
I'm missing the part where the configuredCapacityVector (and configuredMaxCapacityVector) is updated with this information. Are those already have the correctly parsed capacity vectors at the setupQueueConfigs phrase, if so there is no need for parsing them again.

Anyway the field member configurecCapacityVector should have the correct value set somewhere. Some e2e test with JSON asserts should verify it later. Maybe this new UT could also assert it.

Reply:
This is just for validating if the template contained resource vectors, because we need to set weight to 1 if not. The issue here was that we already parsed the capacity vectors we just simply overrode them with "queueContext.getConfiguration().setLabeledQueueWeight(queuePath.getFullPath(), label, 1);"

But I've refactored this, as we don't need to iterate through the node labels this many times.